include/boost/gil/image_processing/threshold.hpp

```diff
@@ -179,6 +181,132 @@ void threshold_truncate
     }
 }
 
+namespace detail{
+
+template <typename SrcView, typename DstView>
+void otsu_impl(SrcView const& src_view, DstView const& dst_view, threshold_direction direction)
+{
+    //deciding output channel type and creating functor
+    typedef typename channel_type<SrcView>::type source_channel_t;
+    typedef typename channel_type<DstView>::type result_channel_t;
+
+    std::array<std::size_t, 256> histData{};
+    source_channel_t min = std::numeric_limits<source_channel_t>::max(),
+        max = std::numeric_limits<source_channel_t>::min();
```

Comment:
Since `source_channel_t` is already displayed in `numeric_limits<source_channel_t>`, this is a good place to use `auto`, e.g.:

```
auto min = std::numeric_limits<source_channel_t>::min()
auto max = std::numeric_limits<source_channel_t>::max()
```

(Regarding min to max and max to min assignment, see my previous comment.)